lenskit-eval/src/main/java/org/lenskit/eval/traintest/predict/PredictEvalTask.java

```diff
@@ -298,7 +298,15 @@ public PredictConditionEvaluator(TableWriter tw, RatingPredictor pred, List<Metr
         public Map<String, Object> finish() {
             Map<String,Object> results = new HashMap<>();
             for (MetricContext<?> mc: predictMetricContexts) {
-                results.putAll(mc.getAggregateMeasurements().getValues());
+                MetricResult measurements = mc.getAggregateMeasurements();
+                // if test and train tests are disjoint we can get "null"
+                if (measurements != null) {
+                    results.putAll(measurements.getValues());
+                } else {
```

Comment:
I'm just talking about the `else` block, not the `if` itself. It's fine for `results` to be missing some values; the NPE is arising when dereferencing `measurements`.

Now, it turns out that `measurements` should never be null under any circumstances, which means one of our metrics has a bug. But this test is good for safety protection against bad metrics.